app/views/plates/_choose_workflow.html.erb

```diff
@@ -1,44 +1,52 @@
-            <%= ro.hidden_field k, value: v %>
+
+  <div id="submission_forms">
+    <% presenter.each_submission_option do |name, submission| %>
+      <%= card do %>
+        <%= form_for submission do |f| %>
+          <%= f.fields_for('request_options') do |ro| %>
+            <% submission.request_options.each do |k,v| %>
+              <%= ro.hidden_field k, value: v %>
+            <% end %>
           <% end %>
-        <% end %>
-        <%= f.hidden_field :template_uuid %>
-        <%= f.fields_for :asset_groups do |asset_groups_form| %>
-          <% submission.asset_groups.each_with_index do |asset_group, group_id| %>
-            <%= asset_groups_form.fields_for group_id.to_s do |ag_form| %>
-              <% asset_group.except(:assets).each do |key, value| %>
-                <%# Deconstruct the other fields of our asset group, such as study and project %>
-                <%= ag_form.hidden_field key, value: value  %>
-              <% end %>
-              <%= ag_form.fields_for :assets do |asset_form| %>
-              <% asset_group[:assets].each do |asset_uuid| %>
-                <%= asset_form.hidden_field nil, value: asset_uuid  %>
-              <% end %>
+          <%= f.hidden_field :template_uuid %>
+          <%= f.fields_for :asset_groups do |asset_groups_form| %>
+            <% submission.asset_groups.each_with_index do |asset_group, group_id| %>
+              <%= asset_groups_form.fields_for group_id.to_s do |ag_form| %>
+                <% asset_group.except(:assets).each do |key, value| %>
+                  <%# Deconstruct the other fields of our asset group, such as study and project %>
+                  <%= ag_form.hidden_field key, value: value  %>
+                <% end %>
+                <%= ag_form.fields_for :assets do |asset_form| %>
+                <% asset_group[:assets].each do |asset_uuid| %>
+                  <%= asset_form.hidden_field nil, value: asset_uuid  %>
+                <% end %>
+                <% end %>
               <% end %>
             <% end %>
           <% end %>
-        <% end %>
 
-        <% if submission.allowed_extra_barcodes %>
-          <%= f.label :extra_barcodes, 'Add additional labware barcodes to the submission (Optional)' %>
-          <br/>
-          <%= f.text_area :extra_barcodes, cols: 30, rows: 5, style: 'width: 100%; resize: none;' %>
-        <% end %>
+          <% if submission.allowed_extra_barcodes %>
+            <%= f.label :extra_barcodes, 'Add additional labware barcodes to the submission (Optional)' %>
+            <br/>
+            <% (1..submission.num_extra_barcodes).each do |num| %>
+              <div class="form-group row">
+                <%= f.label :extra_barcodes, "Barcode #{num}", class:"plate-label col-md-2" %>
+                <%= f.text_field :extra_barcodes, multiple: true, class: "form-control col-md-8" %>
+              </div>
+            <% end %>
+          <% end %>
```

Comment:
![Screenshot 2021-02-05 at 16 10 50](https://user-images.githubusercontent.com/10407809/107058668-ce79a500-67cc-11eb-8bd8-2131b73ebf99.png)

Reply:
Now if you hit tab or enter, it moves onto the next field. Works the same way as the merged_plate page.